### ISSUE TITLE
Get the playing position efficiently

### ISSUE THREAD

**Issue by giohappy** [ricardomatias/ableton-live#8]
AFAIK the only way to get the playing position with Live API is through the `Song.current_song_time` property. It's an observable property (and it's listed inside `SongObservableProperties`) but I fear that it will overload the websocket sending continous updates.

For this (and other use cases) it would be useful to have the ability to observe "custom signals" that could be sent by the Live API device itself.

I was considering to implement an approach similar to "observe" that `ableton-live` already provides:

 - the client asks to observe a custom property ("observecustom" message?). A callback is stored as usual client side
 - the device sends messages for the custom properties (signlas) which the socket might broadcast to any client, or only to clients that requested to observe the custom property

Basically reuse the same mechanism, but without actually using the LiveAPI.

Example for my use case:

 - the device has a `plugsync~`. The outlets are either sent individually or comined in a single message which is sent to the node.script
 - the websocket (for simplicity) boradcasts the message for the custom property (let's call it `playhead`)
 - a client code which asked to observe the playhead by registering it's own callback, receive the `playhead` message

This way:

 - we can extend the Live API device (and maybe use it as a bpatcher in the future?) beyond the Live API itself.
 - we can leverage other Max 4 Live devices that might be optimized or already designed for specific purposes (like `plugsync~`)

@ricardomatias makes sense? Do you suggest alternative solutions?

**Comment by ricardomatias** [comment]
Thanks for the thorough issue @giohappy. The first step is creating a testable scenario where we can see that in fact there are performance issues. Could you create such a test scenario? Only with such a scenario can we start figuring out the origin of the problem and derive solutions.

**Comment by giohappy** [comment]
sure @ricardomatias. I'm back on this personal project and I will do some benchmarking.
